enhancements/network/dpu-ovnkube-operator.md

```diff
@@ -0,0 +1,202 @@
+
+The DPU OVNKube Operator will be responsible for the following host
+configuration on the DPUs which we want to enable hardware offloading:
+
+- Add PF representor to the br-ex bridge
+- Enable switchdev mode on DPU nodes
+- Enable OVS hareware offloading: other_config:hw-offload=true
+- Apply udev rules for BF2 interfaces
+
+We assume a custom MachineConfigPool shall be created for all the DPUs, and the
+DPU OVNKube Operator will also create a MachineConfig CR which injects a systemd
+service to do the aforementioned configuration.
+
+### API
+Example configurations:
+
+```yaml
+apiVersion: smartnic.openshift.io/v1
+kind: OVNKubeConfig
+metadata:
+  name: cluster
+  namespace: tenant-cluster-1
+spec:
+  kubeConfigFile: tenant-cluster-1-kubeconf
+  poolName: bf2-worker
+status:
+  conditions:
+  - lastTransitionTime: ""
+    status: "True"
+    type: MachineConfigApplied
+  - lastTransitionTime: ""
+    status: "True"
+    type: OvnKubeReady
+  - lastTransitionTime: ""
+    status: "False"
+    type: TenantClusterApiServerUnreachable
+```
+
+1. `kubeConfigFile` stores the secret name of the tenant cluster kubeconfig
+   file. The operator uses this to access the api-server of the tenant
+   cluster.
+2. `poolName` specifies the name of the MachineConfigPool CR which contains all
+   the BF2 nodes in the infra cluster. The operator copies the
+   `spec.nodeSelector` of the MCP to render the ovnkube-node daemonset. A new
+   MachineConfig will be created for this MCP, which injects the systemd
+   service.
+
+### Lifecycle Management
+
+The DPU OVNKube Operator will be delivered as an OLM-managed Operator.
+Users can install this operator from the embedded operator hub of OpenShift,
+after the infra cluster installation is complete.
+
+The OLM shall handle the installation, management, upgrade, and uninstallation
+of this operator.
+
+### Test Plan
+
+TBD
+
+### Graduation Criteria
+
+Graduation criteria follows:
+
+#### Dev Preview -> Tech Preview
+
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+#### Tech Preview -> GA
+
+- More testing (upgrade, scale)
+- Add CI job at baremetal OCP CI
+- Sufficient time for feedback
+
+#### Removing a deprecated feature
+N/A
+
+### Upgrade / Downgrade Strategy
+No Kubernetes object is used, hence we could say that it is not
+kubernetes-version sensitive.
+
+The version of this operator shall be tied to the version of the ovn-kubernetes
+running in the tenant cluster.
+
+### Version Skew Strategy
+N/A
+
+## Implementation History
+N/A
+
+## Drawbacks
+N/A
+
+## Alternatives
+
+Implement the above-mentioned function in the Cluster Network Operator. Enable
+this feature when the Cluster Network Operator is running in the infra cluster.
+
+
+## Infrastructure Needed
+
+One X86 bare-metal cluster with BlueField-2 DPU installed on worker nodes, and
+one ARM bare-metal cluster.
```

Comment:
I would explicitly that the ARM bare-metal cluster does not have any separate bare-metal worker nodes.